src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java

```diff
@@ -263,10 +265,25 @@ private Panel getImgViewer() {
 
                 //TODO get metadata and add components
                 Collection<MetadataProperty> metadataProperties = new ArrayList<>();
-                for (int i = 0; i < 5; i++) {
-                    metadataProperties.add(new MetadataProperty<Integer>("property"+i, i, "example property no."+i));
-                }
+                long currentImageId = imageInfo.getImageId();
+                try {
+                    List metadataList = omeroClient.fetchMapAnnotationDataForImage(currentImageId);
+                    for (int i = 0; i < metadataList.size(); i++) {
+                        MapAnnotationData currentMapAnnotation = (MapAnnotationData) metadataList.get(i);
+                        List<NamedValue> list = (List<NamedValue>) currentMapAnnotation
+                            .getContent();
+                        for (NamedValue namedValue : list) {
+                            String metaDataKey = namedValue.name;
+                            String metaDataValue = namedValue.value;
+                            metadataProperties.add(new MetadataProperty<String>("Key: " + metaDataKey, "Value " + metaDataValue, "example property no."));
+                    }
+                    }
 
+                } catch (Exception e) {
+                    LOG.error("Could not retrieve metadata for image:" + currentImageId);
+                    System.out.println(e.toString());
+                    LOG.debug(e);
```

Comment:
```suggestion
                    LOG.error("Could not retrieve metadata for image:" + currentImageId);
                    LOG.debug(e);
                    metadataButton.setEnabled(false);
```

Reply:
Done in seperate commit, good idea🧠  to disable the button if no metadata could be found!

Reply:
If you could somehow disable it right from the beginning and only enable it in case metadata is found this would be even better.

Reply:
Even Better! I added the functionality in 737b119